src/notification.c

```diff
@@ -516,6 +831,31 @@ void notification_init(struct notification *n)
 
 static void notification_format_message(struct notification *n)
 {
+        // Reapply colors based on urgency - this bothered me for quite a long time...
```

Comment:
We can discuss this one in a separate PR/issue as well. I'm not sure if I agree with you here. Two things:
- This approach doesn't work, since the colours are not always stored in the settings struct. They may also be changed by rules and are stored elsewhere
- This is intended behaviour. You are supposed to put your rules that change the urgency above the rules that change the colours based on the urgency.

But feel free to discuss further in a separate PR / issue.

Reply:
Ah, I see - I have the _critical/_normal sections right after global and I append my rules under that. I did not know that this was the intended way to do this.

How would you suggest to do this? Instinctively I expect that if I set urgency to e.g. CRITICAL, it will look like one :slightly_smiling_face:

Perhaps I can re-run rules application after the script change?

Reply:
No, the rules should not be reapplied. The script rule should just work like any other rule. This is way easier to implement. I do agree that it should be documented better in the example dunstrc. In fact, I can do that now.

Note as well that reapplying the rules would create an infinite loop of running scripts and reapplying the rules :p

Reply:
I removed it - will push in few minutes :slightly_smiling_face: As I said, the main source was my misunderstanding of the rules.

After this I would probably suggest to maybe make it a little bit cleaner in the doc. It is obvious now as I know how it works, but initialy from the user perspective I really did not get it :slightly_smiling_face: